### PR TITLE
 CP-29300: Update get_nbd_extents.py to match latest NBD protocol 

### PR DESCRIPTION
**Pull request by gaborigloi**
The protocol has been changed, so that now in the response for
NBD_REPLY_TYPE_BLOCK_STATUS, the last extent can extend beyond the
requested area. The old code would fail with an assertion error in this
case. I've fixed the code to correctly deal with this corner case.

### CODE REVIEW

**Comment by gaborigloi** [comment]
All the quicktests passed

**Comment by gaborigloi** [comment]
The relevant change in the protocol: https://lists.debian.org/nbd/2018/08/msg00005.html

**Comment by thomasmck** [comment]
What testing have you done of the change?

**Comment by gaborigloi** [comment]
I have run the ocaml quicktests that include copying / exporting an empty VDI, a VDI after a random number of random writes, and a VDI filled with random data. These tests verify that the checksum of the result is the same.

File: scripts/get_nbd_extents.py [hunk 1]
```diff
@@ -86,9 +84,15 @@ def _get_extents(path, exportname, offset, length):
 
             # Then process the returned block status info
             assert_protocol(reply['context_id'] == meta_context_id)
+            # Note: There might be consecutive descriptors with the same status value.
             descriptors = reply['descriptors']
-            for descriptor in descriptors:
+            for i, descriptor in enumerate(descriptors):
```
[the remark this answers is not in the document]
Comment (gaborigloi) [reply scripts/get_nbd_extents.py]: I thought it'd be fine to leave it like that, because it's just the index of the descriptor.

**Pull request merged**